File: JS/sort/findNumber.js
"use strict";

const fs = require("fs");

const input =
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString().split("\n")
    : `5
4 1 5 2 3
5
1 3 7 9 5`.split("\n");

const n = parseInt(input[0]);
const arr = input[1].split(" ").map((v) => parseInt(v));

arr.sort((a, b) => a - b);

function binartSearch(x, start, end) {
  const half = parseInt((start + end) / 2);

  if (start > end) return false;

  if (arr[half] === x) return true;
  else {
    if (arr[half] > x) return binartSearch(x, start, half - 1);
    else return binartSearch(x, half + 1, end);
  }
}

const result = Array(parseInt(input[2]));
const findArr = input[3].split(" ").map((v) => parseInt(v));

findArr.forEach((v, i) => {
  binartSearch(v, 0, n - 1) ? (result[i] = 1) : (result[i] = 0);
});

console.log(result.join("\n"));

File: src/sort/FindNumber.java
package sort;

import java.io.*;
import java.util.Arrays;

public class FindNumber {
    static int[] arr;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());
        String[] nums = br.readLine().split(" ");

        arr = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(nums[i]);
        }

        Arrays.sort(arr);

        int m = Integer.parseInt(br.readLine());

        String[] input = br.readLine().split(" ");

        for (int i = 0; i < m; i++) {
            int num = Integer.parseInt(input[i]);

            if(binarySearch(num, 0, n - 1)) bw.write(String.valueOf(1) + "\n");
            else bw.write(String.valueOf(0) + "\n");
        }

        bw.flush();
        bw.close();
    }

    static boolean binarySearch(int x, int start, int end){
        int half = (start + end) / 2;

        if(start > end) return false;
        if(arr[half] == x) return true;
        else if(arr[half] > x){
            return binarySearch(x, start, half - 1);
        }else{
            return binarySearch(x, half + 1, end);
        }
    }
}
